docs/Using-Push-Service.md

```diff
@@ -85,9 +90,34 @@ FirebaseInstanceId.getInstance().instanceId.addOnCompleteListener { task ->
 
 To be able to successfully process notifications, you need to register the app to receive push notifications in the first place. For more information visit [official documentation](https://firebase.google.com/docs/cloud-messaging/android/client).
 
+### Huawei
+For Huawei devices, you can also register your app to receive push notifications using Huawei Push Kit. To integrate Huawei Push Kit into your app, please refer to the Huawei Push Kit documentation.
+
+```kotlin
+// first, retrieve HMS token (do so in the background thread)
+val appId = AGConnectOptionsBuilder().build(context).getString("client/app_id")
+HmsInstanceId.getInstance(context).getToken(appId, "HMS").addOnCompleteListener { task ->
+    if (task.isSuccessful) {
+        val token = task.result
```

Comment:
For Android above, we use `task.result?.token?.let` but here we use `if` - it should be the same, otherwise it's hard to compare differences.

Reply:
hmm... this is my mistake... HMS getToken doesnt return Task<String!> but regular String! So there can't be listener attached to it